jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/WorkspaceConverterMojo.java

```diff
@@ -184,369 +185,374 @@ public void execute() throws MojoExecutionException, MojoFailureException {
-        aPackage.getDevDependencies().stream().collect(Collectors.toMap(Package::getName, Package::getVersion)).forEach(packageJson::addDevDependency);
-        Map<String, String> sortedDependencies = new TreeMap<>();
-        if (packageJson.getDependencies() != null) {
-          packageJson.getDependencies().entrySet().stream()
-                  .sorted(Map.Entry.comparingByKey())
-                  .forEach(entry -> sortedDependencies.put(entry.getKey(), entry.getValue()));
-        }
-        packageJson.setDependencies(sortedDependencies);
-        Map<String, String> sortedDevDependencies = new TreeMap<>();
-        if (packageJson.getDevDependencies() != null) {
-          packageJson.getDevDependencies().entrySet().stream()
-                  .sorted(Map.Entry.comparingByKey())
-                  .forEach(entry -> sortedDevDependencies.put(entry.getKey(), entry.getValue()));
-        }
-        packageJson.setDevDependencies(sortedDevDependencies);
+      PackageJson packageJson = new PackageJson(additionalJsonEntries);
+      packageJson.setName(aPackage.getName());
+      packageJson.setVersion(aPackage.getVersion());
+      if (mavenModule.getData().getOrganization() != null) {
+        packageJson.setAuthor(mavenModule.getData().getOrganization().getName());
+      }
+      packageJson.setLicense("MIT");
+      packageJson.setPrivat(true);
+      aPackage.getDependencies().stream().collect(Collectors.toMap(Package::getName, Package::getVersion)).forEach(packageJson::addDependency);
+      aPackage.getDevDependencies().stream().collect(Collectors.toMap(Package::getName, Package::getVersion)).forEach(packageJson::addDevDependency);
+      Map<String, String> sortedDependencies = new TreeMap<>();
+      if (packageJson.getDependencies() != null) {
+        packageJson.getDependencies().entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .forEach(entry -> sortedDependencies.put(entry.getKey(), entry.getValue()));
+      }
+      packageJson.setDependencies(sortedDependencies);
+      Map<String, String> sortedDevDependencies = new TreeMap<>();
+      if (packageJson.getDevDependencies() != null) {
+        packageJson.getDevDependencies().entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .forEach(entry -> sortedDevDependencies.put(entry.getKey(), entry.getValue()));
+      }
+      packageJson.setDevDependencies(sortedDevDependencies);
+      try {
         FileUtils.write(new File(targetPackageJson), objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(packageJson).concat("\n"));
+      } catch (IOException e) {
+        throw new MojoFailureException(e.getMessage(), e.getCause());
+      }
+    }
+  }
+
+  private void loadAndCopyResource(String fileName) throws IOException {
+    if (Paths.get(convertedWorkspaceTarget + fileName).toFile().exists()) {
```

Comment:
Isn't there a `!` missing? You want to copy the file only if the target _does_ exist?!
I would check whether the file does _not_ exist, then try to copy _without_ REPLACE_EXISTING, then catch the FileAlreadyExists error that can only happen in case a concurrent task already created the file in the tiny time slot between checking and copying. The copying itself is documented to be "atomic".

Reply:
Ok. I will leave a comment why we do this.